### ISSUE TITLE
char* getPythonTraceback(void)   in  scripts/pyplugs.c

### ISSUE THREAD

**Issue by melzaren** [LimpingNinja/nakedmud#2]
I was forced to re-write this, and leaned on AI for the solution. It needs to be properly examined since AI is dumb.

Some resources:
https://docs.python.org/3/c-api/exceptions.html

Note:
void PyErr_Fetch(PyObject **ptype, PyObject **pvalue, PyObject **ptraceback)¶
Part of the Stable ABI.
Deprecated since version 3.12: Use PyErr_GetRaisedException() instead.

void PyErr_NormalizeException(PyObject **exc, PyObject **val, PyObject **tb)
Part of the Stable ABI.
Deprecated since version 3.12: Use PyErr_GetRaisedException() instead, to avoid any possible de-normalization.

So it seems the likely correct solution is to abandon PyErr_Fetch and the normalize tool, and instead use GetRaisedException

**Comment by LimpingNinja** [comment]
This will probably need a conditional:

#if PY_VERSION_HEX >= 0x030C0000  // Python 3.12 or later
   ...
#else
   ...
#endif

I rewrote the <3.12 and wrote the >=3.12 and testing now.

**Comment by melzaren** [comment]
Yes, it didn't even occur to me that I don't even run Python 3.12 (3.11 here). Your approach seems perfectly sensible.
